File: 509. Fibonacci Number/fib.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {
    let [a,b] = [0,1]
    if (n===0) {
        return a
    }
    else if (n===1) {
        return b
    }
    else {
        for (let i=0;i<n-1;i++) {
            [b,a] = [a+b,b]
        }
        return b
    }
};

// Runtime 91 ms / Memory 41.9 MB
// debug: https://jsfiddle.net/

File: 509. Fibonacci Number/fib.cpp
class Solution {
public:
    int fib(int n) {
        int a = 0, b = 1;
        if (n==0) {return a;} //F(0)
        else if (n==1) {return b;} //F(1)
        for ( int i=2; i<n+1; i++ ) { //F(2)~F(n)
            int temp = b;
            b = a+b; //F(n) = F(n-2)+F(n-1)
            a = temp;
        }
        return b;
    }
};